### PR TITLE
created Local v2

### PR DESCRIPTION
**Pull request by Nemo-Git-Hub**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -35,7 +35,9 @@ function App() {
   return (
     <div className="bg-gray-900 h-screen flex items-center justify-center">
       <div className="bg-gray-900 text-gray-200">
-        {data ? <Weather data={data} /> : null}
+        {data ? (
+          <Weather data={data} setLat={setLat} setLong={setLong} />
+        ) : null}
```
Comment (asd-ez) [inline src/App.jsx]: Better use logical AND operator

File: src/components/GooglePlacesAutocomplete.jsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+              }),
+            },
+          }}
+        />
+      </div>
+      <button
+        className="bg-gray-600 p-1 m-1 rounded-lg"
+        onClick={() => {
+          toEdit();
+        }}
```
Comment (asd-ez) [inline src/components/GooglePlacesAutocomplete.jsx]:
```suggestion
        onClick={toEdit}
```

File: src/components/Location.jsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import { BiEditAlt } from "react-icons/bi";
+import { useState } from "react";
+import GooglePlacesSelect from "./GooglePlacesAutocomplete";
+
+function Location({ data, setLat, setLong }) {
+  const [isEdit, setIsEdit] = useState(false);
+
+  function toEdit() {
+    setIsEdit(!isEdit);
+  }
```
Comment (asd-ez) [inline src/components/Location.jsx]: Use arrow function

File: src/components/Location.jsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+  return (
+    <>
+      {isEdit ? (
+        <div className="w-full">
+          <div className="bg-gray-600 p-2 m-2 rounded-lg">
+            <GooglePlacesSelect
+              isEdit={isEdit}
```
Comment (asd-ez) [inline src/components/Location.jsx]:
```suggestion
```

File: src/components/Location.jsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        <>
+          <div className="flex justify-between">
+            <div className="bg-gray-600 p-2 m-2 rounded-lg flex-grow">
+              {data.name}
+            </div>
+            <button
+              className="bg-gray-600 p-2 m-2 rounded-lg"
+              onClick={() => {
+                toEdit();
+              }}
```
Comment (asd-ez) [inline src/components/Location.jsx]:
```suggestion
              onClick={toEdit}
```

File: src/components/GooglePlacesAutocomplete.jsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
+import { useState, useEffect } from "react";
+import GooglePlacesAutocomplete from "react-google-places-autocomplete";
+import { geocodeByAddress, getLatLng } from "react-google-places-autocomplete";
+
+const GooglePlacesSelect = ({ toEdit, setLat, setLong }) => {
+  const [value, setValue] = useState(null); // select place
```
Comment (asd-ez) [inline src/components/GooglePlacesAutocomplete.jsx]: Remove comment or rename with better understand.

File: src/components/GooglePlacesAutocomplete.jsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+  useEffect(() => {
+    if (value) {
+      geocodeByAddress(value.value.description)
```
Comment (asd-ez) [inline src/components/GooglePlacesAutocomplete.jsx]: `value.value` looks strangely, rename `value`

File: src/components/GooglePlacesAutocomplete.jsx [hunk 8]
```diff
@@ -0,0 +1,53 @@
+import { useState, useEffect } from "react";
+import GooglePlacesAutocomplete from "react-google-places-autocomplete";
+import { geocodeByAddress, getLatLng } from "react-google-places-autocomplete";
```
Comment (asd-ez) [inline src/components/GooglePlacesAutocomplete.jsx]: Use 1 import

File: src/components/Weather.jsx [hunk 9]
```diff
@@ -9,18 +10,18 @@ function Weather({ data }) {
           <div className="bg-gray-600 p-2 m-2 rounded-lg text-2xl font-semibold italic">
             {currentDateTime}
           </div>
-          <div className="bg-gray-600 p-2 m-2 rounded-lg">{data.name}</div>
+          <Location data={data} setLat={setLat} setLong={setLong} />
```
Comment (asd-ez) [inline src/components/Weather.jsx]: You always use both functions, maybe better use one function instead?

File: src/components/Location.jsx (hunk 3 above)
Comment (asd-ez) [reply src/components/Location.jsx]: And not use the `isEdit` variable inside `setIsEdit`

File: src/components/Location.jsx (hunk 4 above)
Comment (Nemo-Git-Hub) [reply src/components/Location.jsx]: я как раз думал, что эта строка не нужна)

**Review by asd-ez — APPROVED** [review]

**Comment by asd-ez** [comment]
Add new environment variable for `vercel`, deploy crashed.

**Pull request merged**